com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderUtil.java

```diff
@@ -48,8 +55,27 @@ public static class GLSLCompileResult
         }
 
         public static String stripComments(String source)
-        {
-            return source.replaceAll(regexCommentRemovePattern,"");
+        {  
+            CharStream stream = CharStreams.fromString(source);
+            GLSLLexer lexer = new GLSLLexer(stream);
```

Comment:
It's enough to use lexer here. I think fixing this issue using ANTLR4 is a good starting point for replacing all the regexps with ANTLR lexer and parser.